packages/admin/cms-admin/src/generator/generateForm.ts

```diff
@@ -0,0 +1,358 @@
+        const type = field.type.kind === "NON_NULL" ? field.type.ofType : field.type;
+        return type.kind == "SCALAR" && (type.name == "Float" || type.name == "Int");
+    });
+
+    const out = `
+    import { useApolloClient, useQuery } from "@apollo/client";
+    import {
+        Field,
+        FinalForm,
+        FinalFormInput,
+        FinalFormSaveSplitButton,
+        FinalFormSelect,
+        FinalFormSubmitEvent,
+        MainContent,
+        Toolbar,
+        ToolbarActions,
+        ToolbarFillSpace,
+        ToolbarItem,
+        ToolbarTitleItem,
+        useFormApiRef,
+        useStackApi,
+        useStackSwitchApi,
+        FinalFormCheckbox,
+    } from "@comet/admin";
+    import { ArrowLeft } from "@comet/admin-icons";
+    import { FinalFormDatePicker } from "@comet/admin-date-time";
+    import { BlockState, createFinalFormBlock } from "@comet/blocks-admin";
+    import { EditPageLayout, resolveHasSaveConflict, useFormSaveConflict, queryUpdatedAt } from "@comet/cms-admin";
+    import { CircularProgress, IconButton, FormControlLabel, MenuItem } from "@mui/material";
+    import { FormApi } from "final-form";
+    import { filter } from "graphql-anywhere";
+    import isEqual from "lodash.isequal";
+    import React from "react";
+    import { FormattedMessage } from "react-intl";
+    import { useContentScope } from "@src/common/ContentScopeProvider";
+    import { create${entityName}Mutation, ${instanceEntityName}CheckForChangesQuery, ${instanceEntityName}FormFragment, ${instanceEntityName}FormQuery, update${entityName}Mutation } from "./${entityName}Form.gql";
+    import {
+        GQL${entityName}FormCheckForChangesQuery,
+        GQL${entityName}FormCheckForChangesQueryVariables,
+        GQLCreate${entityName}Mutation,
+        GQLCreate${entityName}MutationVariables,
+        GQL${entityName}FormFragment,
+        GQLUpdate${entityName}Mutation,
+        GQLUpdate${entityName}MutationVariables,
+        GQL${entityName}FormQuery,
+        GQL${entityName}FormQueryVariables,
+    } from "./${entityName}Form.gql.generated";
+    ${Object.entries(rootBlocks)
+        .map(([rootBlockKey, rootBlock]) => `import { ${rootBlock.name} } from "${rootBlock.import}";`)
+        .join("\n")}
+
+    ${
+        Object.keys(rootBlocks).length > 0
+            ? `const rootBlocks = {
+                ${Object.entries(rootBlocks).map(([rootBlockKey, rootBlock]) => `${rootBlockKey}: ${rootBlock.name}`)}
+                };`
+            : ""
+    }
+    
+    type FormValues = ${
+        numberFields.length > 0
+            ? `Omit<GQL${entityName}FormFragment, ${numberFields.map((field) => `"${field.name}"`).join(", ")}>`
+            : `GQL${entityName}FormFragment`
+    } & {
+        ${numberFields.map((field) => `${field.name}: string;`).join("\n")}
+        ${Object.keys(rootBlocks)
+            .map((rootBlockKey) => `${rootBlockKey}: BlockState<typeof rootBlocks.${rootBlockKey}>;`)
+            .join("\n")}
+    };
```

Comment:
If there are no numberFields or rootBlocks, this results in an empty object which eslint doesn't allow. 

```ts
type FormValues = GQLCustomerFormFragment & {};
```
leads to the eslint error 
```
Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.
```

Reply:
fixed